### PR TITLE
Further QA and Quick Fixes

### LINKED ISSUE #217: gripthulu predicts attention marker without any mana 1:00:00

### LINKED ISSUE #333: Blue Out of Range Circle
Out of range circle is sometimes filled with blue. Seems to happen when its not your turn

### PR DESCRIPTION
**Pull request by SoulMuncher**
WIP Draft PR - QA and Fixes for 1.27

Closes #217 
Closes #333 

Fixed Ice/Poison urns pushing units
Fixed Poisoner mana Cost
Re-Adds Gripthulus now that forces are more stable

QA: Sinlgeplayer

### CODE REVIEW

**Comment by SoulMuncher** [comment]
Ready for pull

File: src/Underworld.ts [hunk 1]
```diff
@@ -3277,6 +3277,7 @@ ${CardUI.cardListToImages(player.stats.longestSpell)}
         u.stamina = u.staminaMax;
         const targets = unitSource.getUnitAttackTargets(u, this);
         const canAttack = this.canUnitAttackTarget(u, targets && targets[0])
+        console.log(u, cachedTargets[u.id]?.targets, cachedTargets[u.id]?.canAttack);
```
Comment (jdoleary) [inline src/Underworld.ts:3280]: This needs to be cleaned up

File: src/entity/units/gripthulu.ts [hunk 2]
```diff
@@ -62,34 +63,36 @@ const unit: UnitSource = {
...
-          const pullPromise = forcePushToDestination(chosenUnit, unit, 1, underworld, false);
-          underworld.triggerGameLoopHeadless();
-          await pullPromise;
-        } else {
-          // Only move if not in range
-          const moveTo = math.getCoordsAtDistanceTowardsTarget(unit, chosenUnit, unit.stamina);
-          await Unit.moveTowards(unit, moveTo, underworld);
-        }
-      }
+    console.log("grip targets: ", attackTargets);
```
Comment (jdoleary) [inline src/entity/units/gripthulu.ts:66]: clean up log

File: src/entity/units/poisoner.ts [hunk 3]
```diff
@@ -46,7 +46,7 @@ const unit: UnitSource = {
   action: async (unit: Unit.IUnit, attackTargets: Unit.IUnit[] | undefined, underworld: Underworld, canAttackTarget: boolean) => {
     const chosenUnit = attackTargets && attackTargets[0];
     if (chosenUnit && canAttackTarget) {
-      unit.mana - unit.manaCostToCast;
+      let manaBeforeCast = unit.mana;
```
Comment (jdoleary) [inline src/entity/units/poisoner.ts:49]: oof, I did `-` instead of `-=` lol.  Nice catch

**Review by jdoleary — COMMENTED** [review]
Great changes! Just some logs that need to be cleaned up and it'll be ready to merge.

**Review by jdoleary — APPROVED** [review]

**Pull request merged**